app/models/generic_event.rb

```diff
@@ -13,17 +13,13 @@ class GenericEvent < ApplicationRecord
     eventable_type
     details
   ].freeze
-  STI_CLASSES = %w[
-    MoveCancel
-    JourneyCancel
-    JourneyComplete
-    JourneyLockout
-    JourneyLodging
-    JourneyReject
-    JourneyStart
-    JourneyUncancel
-    JourneyUncomplete
-  ].freeze
+
+  STI_CLASSES = Dir['app/models/generic_event/*'].map { |file|
+    file
+      .gsub('app/models/generic_event/', '')
+      .sub('.rb', '')
+      .camelcase
+  }.freeze
```

Comment:
IMO this makes things hard to understand, and less explicit.
but I like the code, I'd move it into the test.

Reply:
Have updated 👍